source/state_representation/src/parameters/ParameterInterface.cpp

```diff
@@ -4,14 +4,16 @@ namespace state_representation {
 
 ParameterInterface::ParameterInterface(
     const std::string& name, const ParameterType& type, const StateType& parameter_state_type
-) : State(name), parameter_type_(type), parameter_state_type_(parameter_state_type) {
-  this->set_type(StateType::PARAMETER);
-}
+) : State(StateType::PARAMETER, name), parameter_type_(type), parameter_state_type_(parameter_state_type) {}
 
-ParameterInterface::ParameterInterface(const ParameterInterface& parameter) :
-    State(parameter),
-    parameter_type_(parameter.get_parameter_type()),
-    parameter_state_type_(parameter.get_parameter_state_type()) {}
+//ParameterInterface::ParameterInterface(const ParameterInterface& parameter) :
```

Comment:
don't forget to remove commented blocks if you're happy with it. Default copy seems fine by me